Add context help

corviday opened the pull request:
This PR adds a bunch of help popups to the UI. Please suggest:
* other elements that need UI popups
* improved wording

[Demo](https://services.pacificclimate.org/dev/scip/app/)

resolves #61 
resolves #74

rod-glover reviewed (changes requested):
This is a real improvement in the usability and clarity of the interface. Well done.

Feedback for (possible) change:

1. I think the info icons would look better if they were spaced away from the text they are next to.
2. Adjust their vertical spacing so that they are centered on the text line?

Other remarks:
1. I wonder if the instructional text at the bottom of the controls ("Select a region either using the circle marker tool  ...") could be placed in an info marker somewhere. You'd gain some much-needed space below the map.
3. Part of that text presently reads "from the dropdown watershed and conservation unit menus below". I think that should be "above". Of course, if you put this into an info popup, then that might not change, depending on placement.
4. Behavioural problem, I think:
   1. Refresh app.
   2. Select a basin.
   3. Select a  watershed.
   4. Select a yearly (or monthly or daily) indicator.
   5. "Loading data" message shows. The data does not seem ever to load.
   6. Click View Indictors at outlet only switch. (Also note spelling mistake "Indictors").
   7. Loading data message immediately replaced by graph.
5. More generally and I think more misleading:
   1. Switch "View indicators ..." back off.
   2. Select a different indicator.
   3. Graph does not update, but the title does (!)
   4. Switch "View indicators ..." on.
   5. Graph updates.

corviday commented on public/help.yaml:
```diff
@@ -0,0 +1,40 @@
+    body: |
+        These controls change which indicator data is displayed on the map. 
+        The outermost set of buttons change the displayed climatology. There are four climatologies available: historical, early century, mid century, and end of century.
+        The innermost set of buttons change the displayed timestamp for daily or monthly indicators. They are disabled for annual indicators.
```
thank you, this is extremely helpful!

corviday commented:
Updated [demo](https://services.pacificclimate.org/dev/scip/app/) incorporating suggestions and including the help pdf.

I also removed three buttons we weren't using from the graph mouseover menu.

rod-glover commented on src/components/InfoPopup/InfoPopup.js:
```diff
@@ -0,0 +1,49 @@
+        key={`${index}-help-popup`}
+        placement={placement}
+        overlay={
+          <Popover id={helpText[index] ? helpText[index].title : "Loading"}>
```
Expressions like this can get shorter and easier to read by using the [optional chaining operator `?.`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) and the [nullish coalescing operator `??`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) (terrible name; means well-behaved version of `||`). 

This one would become `helpText[index]?.title ?? "Loading"`. 

A really convenient thing about `?.` is that it can be used repeatedly in the same expression: `a?.b.?.c?.d`, and it returns undefined if any prop doesn't exist. So much better than the equivalent ternary `?` expression.

rod-glover commented on src/components/App/App.js:
```diff
@@ -49,12 +53,15 @@ function App() {
           <h1>Salmon Climate Impacts Portal</h1>
         </div>
+        <div className="Help">
+            <a href={`${process.env.PUBLIC_URL}/general_help.pdf`}>General Help</a>
```
This file appears to be missing from `./public/` and the link currently does nothing. 

Also maybe add `target="_blank"` attribute so that it opens in a new tab?

rod-glover commented on src/components/AreaDisplay/AreaDisplay.js:
```diff
@@ -260,9 +260,7 @@ function collateRegions(regions, whitelist) {
-      marker and select everything upstream of it, or from the dropdown watershed and
-      conservation unit menus below. You can narrow down regions shown on the menus
-      by river basin or species.
+      marker and select everything upstream of it or by name from the dropdown menus above.
```
Nice, way tighter.

rod-glover reviewed (approved):
Nice, all good except for missing General Help PDF.

This is sort of help-related: On the Daily Indicators tab, there are two drop-downs, the indicator variable and the time period.
- Does the time period need a label and info popup?
- Why isn't there a time period selector for Yearly and Monthly Indicators?

corviday commented:
> * Does the time period need a label and info popup?

Can't hurt to give it one.

> * Why isn't there a time period selector for Yearly and Monthly Indicators?

They show all climatologies at once. This is less feasible for the "daily" graphs with their 365 individual points, way more work for the backend than the monthly or annual ones.
